### PR TITLE
[CIS-1578] Automate command line actions when merging release

### PR DESCRIPTION
**Pull request by alteral**
### 🔗 Issue Links

- https://stream-io.atlassian.net/browse/CIS-1578

### 🎯 Goal

- Create command line scripts/lanes to leave no room for error when release is merged.

### 📝 Summary

There will be two new fastlane lanes:
- `bundle exec fastlane merge_release_to_main`
- `bundle exec fastlane merge_main_to_develop`

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

gif

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -110,6 +110,28 @@ lane :release do |options|
   UI.success("Successfully started release #{version_number}")
 end
 
+lane :merge_release_to_main do
+  ensure_git_status_clean
+  sh('git checkout main')
+  sh('git pull')
+  release_branches = sh(command: 'git branch -a', log: false).delete(' ').split("\n").grep(%r(origin/.*release/))
+  UI.user_error!("Expected 1 release branch, found #{release_branches.size}") if release_branches.size != 1
+  sh("git merge #{release_branches.first} --ff-only") # ff-only is important. Read: https://notion.so/iOS-Branching-Strategy-37c10127dc26493e937769d44b1d6d9a
```
Comment (nuno-vieira) [inline fastlane/Fastfile:119]: What is this part actually doing? Maybe you can provide a comment? How does it know it is the most recent release?

Comment (alteral) [reply fastlane/Fastfile:119]: Actually, it does not know anything about release branches apart from their cound. It searches for all `release` branches in `origin` and:
- Raises a error if there are more than one branch
- Picks the branch name of there is only one branch
 
As per our workflow we always have only one release branch, I thought it is safe to do it this way.

If you think it's not the ideal option, I can provide a `prompt` if there is more than one branch so the release engineer could pick the correct one manually. Wdyt?

Comment (alteral) [reply fastlane/Fastfile]: Done. Check it out 🙂

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**